Description: adjust sed regexps for bison 3.0's changed whitespace
Author: Faidon Liambotis <paravoid@debian.org>
Last-Update: 2014-03-05
 make-parser.sh does some evil post-processing on bison's C output and these
 matches are hence dependent on the C/preprocessor code emitted by bison. It
 looks like that between bison 2.0.5 that upstream uses and bison 3.0, the
 indentation(!) changed, making those grep/seds to not match. Adjust these,
 using \s+ to make them compatible with both versions.

--- a/hphp/parser/make-parser.sh
+++ b/hphp/parser/make-parser.sh
@@ -38,13 +38,13 @@ fi
 # enum to a series of macro invocations, and then find the first and last macro
 # invocations.
 $SED -i -r -e 's/(T_\w+) = ([0-9]+)\s*,?/YYTOKEN(\2, \1)/g' \
-     -e "s/   enum yytokentype/#ifndef YYTOKEN_MAP\n#define YYTOKEN_MAP enum yytokentype\n#define YYTOKEN(num, name) name = num,\n#endif\n   YYTOKEN_MAP/" \
+     -e "s/\s*enum yytokentype/#ifndef YYTOKEN_MAP\n#define YYTOKEN_MAP enum yytokentype\n#define YYTOKEN(num, name) name = num,\n#endif\n   YYTOKEN_MAP/" \
     ${OUTHEADER}
 
-cat ${OUTHEADER} | grep "     YYTOKEN(" | head -n 1 | \
-    $SED -r -e 's/     YYTOKEN.([0-9]+).*/#ifndef YYTOKEN_MIN\n#define YYTOKEN_MIN \1\n#endif/' >> ${OUTHEADER}
-cat ${OUTHEADER} | grep "     YYTOKEN(" | tail -n 1 | \
-    $SED -r -e 's/     YYTOKEN.([0-9]+).*/#ifndef YYTOKEN_MAX\n#define YYTOKEN_MAX \1\n#endif/' >> ${OUTHEADER}
+cat ${OUTHEADER} | grep "^ \+YYTOKEN(" | head -n 1 | \
+    $SED -r -e 's/\s+YYTOKEN.([0-9]+).*/#ifndef YYTOKEN_MIN\n#define YYTOKEN_MIN \1\n#endif/' >> ${OUTHEADER}
+cat ${OUTHEADER} | grep "^ \+YYTOKEN(" | tail -n 1 | \
+    $SED -r -e 's/\s+YYTOKEN.([0-9]+).*/#ifndef YYTOKEN_MAX\n#define YYTOKEN_MAX \1\n#endif/' >> ${OUTHEADER}
 
 # Renaming some stuff in the cpp file
 $SED -i \
